caffe2/CMakeLists.txt

```diff
@@ -653,6 +653,7 @@ if(USE_CUDA)
   )
   set_source_files_properties(${TORCH_SRC_DIR}/csrc/jit/passes/frozen_conv_add_relu_fusion.cpp PROPERTIES COMPILE_FLAGS "-DUSE_CUDA=1")
   set_source_files_properties(${TORCH_SRC_DIR}/csrc/jit/codegen/cuda/interface.cpp PROPERTIES COMPILE_FLAGS "-DUSE_CUDA=1")
+  set_source_files_properties(${TORCH_SRC_DIR}/csrc/autograd/engine.cpp PROPERTIES COMPILE_FLAGS "-DUSE_CUDA=1")
```

Comment:
Why isn't this set for all files already. That sounds like a bug if it is not?

Reply:
Apparently `USE_CUDA` was not set. Or could it be that `defined` function in `#if defined(USE_CUDA)` is misbehaving? All I see that the code never checks the existence of primary context.

Reply:
Confirmed, USE_CUDA is just not being defined.

Reply:
No, I don't, sorry. There was this dream that we can have device-agnostic pieces of code, and thus not have any ifdefs in them, like engine.cpp, but apparently this doesn't work.
This would also need corresponding changes in the internal builds, cc @dagitses